src/sentry/api/helpers/group_index/index.py

```diff
@@ -77,6 +79,34 @@ def build_query_params_from_request(
         except ValueError:
             raise ParseError(detail="Invalid cursor parameter.")
     query = request.GET.get("query", "is:unresolved").strip()
+    if request.GET.get("savedSearch") == "0":
+        saved_searches = (
+            SavedSearch.objects
+            # Do not include pinned or personal searches from other users in
+            # the same organization. DOES include the requesting users pinned
+            # search
+            .exclude(
+                ~Q(owner_id=request.user.id),
+                visibility__in=(Visibility.OWNER, Visibility.OWNER_PINNED),
+            )
+            .filter(
+                Q(organization=organization) | Q(is_global=True),
+            )
+            .extra(order_by=["name"])
+        )
+        selected_search_id = request.GET.get("searchId", None)
+        if selected_search_id:
+            # saved search requested by the id
+            saved_search = saved_searches.filter(id=int(selected_search_id))
```

Comment:
if we use .first higher up we can skip .exists which makes an additional sql query
```suggestion
            saved_search = saved_searches.filter(id=int(selected_search_id)).first()
```